torch/distributed/_composable/fsdp/_fsdp_param_group.py

```diff
@@ -309,7 +312,8 @@ def post_backward(self, *unused: Any):
         self._training_state = TrainingState.POST_BACKWARD
         with torch.profiler.record_function("FSDP::post_backward_reshard"):
             if not self.reduce_grads:
-                self.reshard()
+                if self.reshard_after_backward:
```

Comment:
This is the main expected use case: do not reduce gradients, and do not reshard parameters.

Reply:
is the use case mainly for pipeline parallel?

Reply:
This is useful for any kind of gradient accumulation, where pipeline parallelism is one example.